### PR TITLE
Some cleanup and adding more tests (May 2016)

### PR DESCRIPTION
**Pull request by drmarkwslater**
Adding a little a bit of cleanup and converting some more of the old tests. This also includes some minor doc changes for the Condor backend to make things clearer and some clean up of `Ganga.Core.__init__.py` - just some rearrangement and removal of unused code.

### CODE REVIEW

**Comment by drmarkwslater** [comment]
Apologies - forgot to change the PR title from the branch title. Fixed and made the PR comment a bit clearer as well!

**Comment by drmarkwslater** [comment]
I believe this can also be merged for 6.3.1 or 6.4. Any objections?

**Comment by mesmith75** [comment]
Do we want to merge this now?

File: python/Ganga/Utility/Decorators.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+def static_vars(**kwargs):
+    """
+    Decorate a function and provide it with some static variables.
+    >>> @static_vars(counter=0)
```
Comment (alexanderrichards) [inline python/Ganga/Utility/Decorators.py]: actually this is a doctest

File: python/Ganga/test/GPI/Bugs/TestSavannah10016.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from Ganga.testlib.GangaUnitTest import GangaUnitTest
+
+
+class TestSavannah10016(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(TestSavannah10016, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
```
Comment (alexanderrichards) [inline python/Ganga/test/GPI/Bugs/TestSavannah10016.py]: Can this not be at the top of the module?

Comment (egede) [inline python/Ganga/test/GPI/Bugs/TestSavannah10016.py]: @alexanderrichards Done

**Pull request merged**